tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -795,9 +801,67 @@ def _post_fit(self, keep_only_best=False, refit_full=False, set_best_to_refit_fu
         if keep_only_best:
             self.delete_models(models_to_keep='best', dry_run=False)
 
+        if calibrate and self.problem_type in PROBLEM_TYPES_CLASSIFICATION:
+            self._calibrate_model()
+        else:
+            logger.log(30, 'WARNING: calibrate is only applicable to classification problems')
+
         if save_space:
             self.save_space()
 
+    def _calibrate_model(self, model_name: str = None, lr: float = 0.01, max_iter: int = 1000, init_val: float = 1.0):
+        """
+        Applies temperature scaling to the best autogluon model. Applies
+        inverse softmax to predicted probs then trains temperature scalar
+        on validation data to maximize negative log likelihood. Inversed
+        softmaxes are divided by temperature scalar then softmaxed to return
+        predicted probs
+
+        Parameters:
+        -----------
+        model_name: str: default=None
+            model name to retreive from trainer and fit
+        """
+        if model_name is None:
+            model_name = self._trainer.get_model_best()
+
+        if self._trainer.bagged_mode:
+            y_val_probs = self.get_oof_pred_proba(model_name).to_numpy()
+            y_val = self._trainer.load_y().to_numpy()
+        else:
+            X_val = self._trainer.load_X_val()
+            y_val_probs = self._trainer.predict_proba(X_val, model_name)
+            y_val = self._trainer.load_y_val().to_numpy()
+
+        if self.problem_type == BINARY:
+            y_val_probs = LabelCleanerMulticlassToBinary.convert_binary_proba_to_multiclass_proba(y_val_probs)
+
+        try_import_torch()
+        import torch
+
+        y_val_tensor = torch.tensor(y_val)
+        temperature_param = torch.nn.Parameter(torch.ones(1).fill_(init_val))
+        logits = torch.tensor(np.log(y_val_probs))
+        nll_criterion = torch.nn.CrossEntropyLoss()
+        optimizer = torch.optim.LBFGS([temperature_param], lr=lr, max_iter=max_iter)
+        scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=0.99)
```

Comment:
Why is this a different implementation than in abstract_model? Consider moving the logic to a function and re-using it here and in abstract model

Reply:
nevermind, I mis-read. The logic should still be in a function though.